File: app/javascript/channels/room_channel.js
import consumer from "./consumer"

const appRoom = consumer.subscriptions.create("RoomChannel", {
  
  connected() {
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    const Messages = document.getElementById('messages');
    Messages.insertAdjacentHTML('beforeend', data['message']);
  },

  speak: function(message, room_id) {
    return this.perform('speak', { message: message,room_id: room_id });
  }
});

if(/rooms/.test(location.pathname)) {
  $(document).on("keydown", ".room__message-form_textarea", function(e) {
    if (e.key === "Enter") {
      const room_id = $('textarea').data('room_id')
      appRoom.speak(e.target.value,room_id);
      e.target.value = '';
      e.preventDefault();
    }
  })
}

File: app/channels/room_channel.rb
class RoomChannel < ApplicationCable::Channel
  def subscribed
    stream_from "room_channel_8"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def speak(data)
    
    Message.create!(
      content: data['message'],
      user_id: current_user.id,
      room_id: data['room_id']
    )
  end
end
